### PR TITLE
Feat: Out of office autoresponder

### LINKED ISSUE #2035: Feat: Out of office autoresponder
Hi, I just updated chatwoot to the newest version and I can see that the UI for working hours is here :) That is awesome, thanks!!

I noticed that after the feature became available for everyone, it caused some problems such as those in #1827 & #1824 that were fixed in #1842. I notice that there is still a comment to remove the part of the code that I put there, so I think I'm a good person to explain what it was doing (please, don't hesitate to mention me @azyzio if something isn't clear)

https://github.com/chatwoot/chatwoot/blob/develop/app/services/message_templates/hook_execution_service.rb#L8
```ruby
    # TODO: let's see whether this is needed and remove this and related logic if not
    # ::MessageTemplates::Template::OutOfOffice.new(conversation: conversation).perform if should_send_out_of_office_message?
```

The above was responsible for sending automatic out-of-office messages. I can see that currently, the out-of-office message appears in the window before starting the conversation, but nothing really happens when someone does write to us:
image
image

The behavior I designed was that the out-of-office message was sent as a response to someone trying to reach out after working hours (something like a chatbot response). I think it makes a lot of sense as some customers might not read the note when starting the conversation. Out-of-office message is there to manage their expectations and send them to some self-service solutions. We can't do it after disabling the feature. Here is the example from Tidio chat app.
2b057083-d6ba-4c4f-8cde-2cd856400ac2

I am ready to fix all the bugs that were caused by the out-of-office autoresponder. Just please let me know that the described behavior is something you are willing to re-introduce :)

@pranavrajs @sojan-official

### PR DESCRIPTION
**Pull request by aswindevps**
# Pull Request Template

## Description

This change allows the user to enable autoresponder during the out-of-office time.

Fixes #2035

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

1. Verified the business hour configuration in different channels.
2. Autoresponder messages get sent correctly when a message comes in outside office hours for the website channel.
3. Couple of tests were added for the **working_hour** modal.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by aswindevps** [comment]
Screenshot 2021-09-10 at 3 32 29 PM
Screenshot 2021-09-10 at 3 34 02 PM

File: spec/models/working_hour_spec.rb [hunk 1]
```diff
@@ -26,4 +26,28 @@
...
   end
+
+  context 'when on friday 12:30pm' do
+    before do
+      Time.zone = 'UTC'
+      create(:working_hour)
+      travel_to '10.09.2021 12:30'.to_datetime
+    end
+
+    it 'is considered out of office' do
```
Comment (sojan-official) [inline spec/models/working_hour_spec.rb:37]:
```suggestion
    it 'is considered to be in business hours' do
```

**Comment by sojan-official** [comment]
let enable back the specs in  `hook_execution_service_spec` which should add additional coverage https://github.com/chatwoot/chatwoot/blob/a2e05f54e34fa18a0dad3d2ab2483895216f3f4d/spec/services/message_templates/hook_execution_service_spec.rb#L157

File: spec/models/working_hour_spec.rb (hunk 1 above)
Comment (aswindevps) [reply spec/models/working_hour_spec.rb]: Done

**Review by sojan-official — DISMISSED** [review]
Code LGTM. let's QA and get this merged

File: app/services/message_templates/hook_execution_service.rb [hunk 2]
```diff
@@ -15,13 +15,16 @@ def perform
 
   def trigger_templates
     # TODO: let's see whether this is needed and remove this and related logic if not
```
Comment (sojan-official) [inline app/services/message_templates/hook_execution_service.rb:17]:
```suggestion
   
```

let's remove the comment as it don't make sense anymore

File: spec/services/message_templates/hook_execution_service_spec.rb [hunk 3]
```diff
@@ -154,25 +154,40 @@
...
+      # described class gets called in message after commit
+      message = create(:message, conversation: conversation)
+
+      expect(::MessageTemplates::Template::OutOfOffice).to have_received(:new).with(conversation: message.conversation)
+      expect(out_of_office_service).to have_received(:perform)
+    end
+
+    it 'will not call ::MessageTemplates::Template::OutOfOffice if its a tweet conversation' do
+      twitter_channel = create(:channel_twitter_profile)
+      twitter_inbox = create(:inbox, channel: twitter_channel)
```
Comment (sojan-official) [inline spec/services/message_templates/hook_execution_service_spec.rb:179]: let's enable office hours on this inbox as well for the test to work correctly 

twitter_inbox.update(working_hours_enabled: true, out_of_office_message: 'We are out of office')

File: app/services/message_templates/hook_execution_service.rb (hunk 2 above)
Comment (aswindevps) [reply app/services/message_templates/hook_execution_service.rb]: Removed

File: spec/services/message_templates/hook_execution_service_spec.rb (hunk 3 above)
Comment (aswindevps) [reply spec/services/message_templates/hook_execution_service_spec.rb:179]: Enabled office hours for twitter inbox in test

**Pull request merged**